File: sukh_1684/problems/valid_anagram/solution.java
class Solution {
    public boolean isAnagram(String s, String t) {
        int len1=s.length();
        int len2= t.length();
        if(len1!= len2){
            return false;
        }
           HashMap<Character,Integer> check = new HashMap<>();
          for(int i=0;i<len1;i++){
              char c = s.charAt(i);
              check.put(c,check.getOrDefault(c,0)+1);
          }
        for(int i=0;i<len2;i++){
            char c = t.charAt(i);
            if(!check.containsKey(c)){
                return false;
            }
            int temp = check.get(c)-1;
            if(temp == -1){
                return false;
            }
            check.put(c,temp);
        }
        
        
        return true;
    }
}

File: sukh_1684/problems/valid_anagram/solution.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        sN = len(s)
        tN = len(t)
        if sN != tN:
            return False
        s = ''.join(sorted(s))
        t = ''.join(sorted(t))
        for i in range(sN):
            if s[i] != t[i]:
                return False
        return True
